routing/speed_camera_manager.hpp

```diff
@@ -0,0 +1,168 @@
+#pragma once
+
+#include "routing/route.hpp"
+#include "routing/routing_callbacks.hpp"
+#include "routing/routing_helpers.hpp"
+#include "routing/speed_camera.hpp"
+#include "routing/turns_notification_manager.hpp"
+
+#include "platform/location.hpp"
+
+#include "base/assert.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <queue>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace routing
+{
+// Do not touch the order, it uses in platforms.
+enum class SpeedCameraManagerMode
+{
+  Auto,
+  Always,
+  Never,
+
+  MaxValue
+};
+
+class SpeedCameraManager
+{
+public:
+  static std::string const kSpeedCamModeKey;
+
+  explicit SpeedCameraManager(turns::sound::NotificationManager & notificationManager);
+
+  void SetRoute(std::weak_ptr<Route> route) { m_route = std::move(route); }
+
+  void SetSpeedCamShowCallback(SpeedCameraShowCallback && callback)
+  {
+    m_speedCamShowCallback = std::move(callback);
+  }
+
+  void SetSpeedCamClearCallback(SpeedCameraClearCallback && callback)
+  {
+    m_speedCamClearCallback = std::move(callback);
+  }
+
+  bool Enable() const { return m_mode != SpeedCameraManagerMode::Never; }
+
+  void OnLocationPositionChanged(location::GpsInfo const & info);
+
+  void GenerateNotifications(std::vector<std::string> & notifications);
+  bool MakeBeepSignal();
+
+  void Reset();
+
+  void SetMode(SpeedCameraManagerMode mode)
+  {
+    m_mode = mode;
+    settings::Set(kSpeedCamModeKey, static_cast<int>(mode));
+  }
+
+  SpeedCameraManagerMode GetMode() const { return m_mode; }
+
+private:
+  enum class Interval
+  {
+    // Influence zone of camera, set by |kInfluenceZoneMeters| const
```

Comment:
`// Influence zone of camera, set by |kInfluenceZoneMeters|.`

Reply:
done